### PR TITLE
test(html): add missing edge-case and webpackIgnore tests for html-loader parity

### PR DESCRIPTION
**Pull request by aryanraj45**
**Summary**

  Audit of html-loader feature parity revealed several edge cases not covered by
  existing tests. Adds a new `edge-cases` configCase and extends two existing cases to
  close the gaps.

  **What kind of change does this PR introduce?**

  test

  **Did you add tests for your changes?**

  Yes — `test/configCases/html/edge-cases/` (new), `test/configCases/html/link/`
  (extended), `test/configCases/html/modulepreload-esm/` (extended).

  **Does this PR introduce a breaking change?**

  No.

  **If relevant, what needs to be documented once your changes are merged or what have 
  you already documented?**

  n/a
**Use of AI**
partial

### CODE REVIEW

File: test/configCases/html/edge-cases/__snapshots__/ConfigTest.snap [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+exports[`ConfigTestCases html edge-cases exported tests should skip non-resolvable URLs and match tags case-insensitively 1`] = `
+"<!DOCTYPE html>
+<html>
+<head><title>Edge cases</title></head>
+<body>
+
+<!-- Empty value: no dependency, left as-is -->
+<img src=\\"\\" alt=\\"empty\\">
+<link rel=\\"icon\\" href=\\"\\">
+<img srcset=\\"\\" alt=\\"empty-srcset\\">
```
[the remark this answers is not in the document]
Comment (aryanraj45) [reply test/configCases/html/edge-cases/__snapshots__/ConfigTest.snap:12]: In a .snap file, \\" = one literal ".So href=\\"\\" in the snap file = href="" in actual output-which is exactly correct. The <link rel="icon" href=""> input has an empty href, the serializer leaves it as-is, and the snapshot records that faithfully.

File: test/configCases/html/edge-cases/__snapshots__/ConfigCacheTest.snap [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+exports[`ConfigCacheTestCases html edge-cases exported tests should skip non-resolvable URLs and match tags case-insensitively 1`] = `
+"<!DOCTYPE html>
+<html>
+<head><title>Edge cases</title></head>
+<body>
+
+<!-- Empty value: no dependency, left as-is -->
+<img src=\\"\\" alt=\\"empty\\">
+<link rel=\\"icon\\" href=\\"\\">
+<img srcset=\\"\\" alt=\\"empty-srcset\\">
```
[the remark this answers is not in the document]
Comment (aryanraj45) [reply test/configCases/html/edge-cases/__snapshots__/ConfigCacheTest.snap:12]: same

File: test/configCases/html/modulepreload-esm/index.js [hunk 3]
```diff
@@ -24,6 +24,11 @@ it("should rewrite <link rel=modulepreload> and <script type=module src> to chun
 	expect(page).toMatch(/<script type="module" src="__html_[^"]+\.mjs">/);
 	// Data URI in modulepreload is also bundled.
 	expect(page).not.toContain('href="data:text/javascript');
+	// webpackIgnore leaves modulepreload and module-script entries untouched.
+	expect(page).toContain(
+		'<link rel="modulepreload" href="./ignored-preload.js">'
+	);
+	expect(page).toContain('<script type="module" src="./ignored-entry.js">');
```
[the remark this answers is not in the document]
Comment (aryanraj45) [reply test/configCases/html/modulepreload-esm/index.js:31]: The existing assertions in this file follow a consistent pattern of using `toContain` with partial string matches (see lines 20–26, pre-PR). The two newly added assertions adhere to that same convention. Switching to assertions on the full closing tag would introduce inconsistency with the surrounding code without providing any meaningful additional safety, since the snapshot already validates the complete serialized output.
